File: Twitter_Bot/lambda_twitter_bot.py
import json
import tweepy
import boto3
from botocore.exceptions import ClientError

# Retrieve the Twitter API keys and tokens from AWS Secrets Manager
def get_secret():

    secret_name = "your_secret_name"
    region_name = "your_region_name"

    # Create a Secrets Manager client
    session = boto3.session.Session()
    client = session.client(
        service_name='secretsmanager',
        region_name=region_name
    )

    try:
        get_secret_value_response = client.get_secret_value(
            SecretId=secret_name
        )
    except ClientError as e:
        # For a list of exceptions thrown, see
        # https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_GetSecretValue.html
        raise e

    # Decrypts secret using the associated KMS key.
    secret = get_secret_value_response['SecretString']
    return secret


# Get the Twitter API secrets from AWS Secrets Manager
secrets = json.loads(get_secret())
consumer_key = secrets['your_consumer_key']
consumer_secret = secrets['your_consumer_secret']
bearer_token = secrets['your_bearer_token']
access_token = secrets['your_access_token']
access_token_secret = secrets['your_access_token_secret']
client_id = secrets['your_client_id']
client_secret = secrets['your_client_secret']

# Authenticate to the Twitter API
auth = tweepy.OAuthHandler(consumer_key, consumer_secret)
auth.set_access_token(access_token, access_token_secret)
api = tweepy.API(auth)


def lambda_handler(event, context):
    try:
        # Search for recent tweets from the user @Aztec_MBB
        tweets = api.user_timeline(screen_name="your_twitter_handle")

        # Like and retweet each tweet
        for tweet in tweets:
            api.create_favorite(tweet.id)
            api.retweet(tweet.id)
            print("Liked and retweeted tweet:", tweet.id)
    except tweepy.TweepError as error:
        print("Error:", error)

File: Twitter_Bot/lambda_twitter_bot.js
const AWS = require('aws-sdk');
const twitter = require('twitter');

// Retrieve the Twitter API keys and tokens from AWS Secrets Manager
async function getSecret() {
  const secretName = 'your_secret_name';
  const regionName = 'your_region_name';

  // Create a Secrets Manager client
  const client = new AWS.SecretsManager({ region: regionName });

  try {
    const result = await client.getSecretValue({ SecretId: secretName }).promise();
    return JSON.parse(result.SecretString);
  } catch (err) {
    throw err;
  }
}

// Authenticate to the Twitter API
async function main() {
  try {
    const secrets = await getSecret();
    const client = new twitter({
      consumer_key: secrets.your_consumer_key,
      consumer_secret: secrets.your_consumer_secret,
      bearer_token: secrets.your_bearer_token,
    });

    // Search for recent tweets from the user @Aztec_MBB
    const tweets = await client.get('statuses/user_timeline', { screen_name: 'your_twitter_handle' });

    // Like and retweet each tweet
    for (const tweet of tweets) {
      await client.post('favorites/create', { id: tweet.id_str });
      await client.post('statuses/retweet', { id: tweet.id_str });
      console.log(`Liked and retweeted tweet: ${tweet.id_str}`);
    }
  } catch (err) {
    console.error(err);
  }
}

main();
